### ISSUE TITLE
Fix Data Leakage Issue: Separate Training and Validation by Time

### ISSUE THREAD

**Issue by mlaurako** [NOAA-EMC/ocelot#49]
Our current training setup has data leakage between training and validation sets, where validation data is temporally adjacent to training data within the same time window.

### Original Method

1. Data is sampled randomly or sequentially within full_start_date and full_end_date
2. Random sampling picks a start date, then extracts a window (e.g., 14 days) sequentially
3. This window is processed into time bins, then input-target pairs
4. First ~80% of pairs become training data, remaining ~20% become validation data

**Problem**: Validation data is always temporally adjacent to training data, and validation data from one epoch can become training data in another epoch.

### Proposed Data Split
When we have multiple years of data, we can separate them by year (e.g., 5 years training, 1 year validation, 1 year testing). Currently, our v5 dataset contains **3 months** of data. The proposed split is:

* Training: 10 weeks (~90%)
* Validation: 2 weeks (~10%)
* Testing: None (until we expand our dataset)

### Proposed Code Update
1. Update `train_gnn.py`

* Add `TRAIN_VAL_SPLIT_RATIO` parameter (default: 0.9)
* Calculate separate training and validation date ranges:
```
TRAIN_START_DATE = FULL_START_DATE
TRAIN_END_DATE = FULL_START_DATE + (total_days * TRAIN_VAL_SPLIT_RATIO)
VAL_START_DATE = TRAIN_END_DATE  
VAL_END_DATE = FULL_END_DATE
```

2. Update `callbacks.py`
* `on_train_epoch_start`: Sample training windows only from `[TRAIN_START_DATE, TRAIN_END_DATE]`
* `on_validation_epoch_start`: Sample validation windows only from `[VAL_START_DATE, VAL_END_DATE]`

3. Update `gnn_datamodule.py`
* Remove internal train/val splitting logic in `setup()`
* Use ALL bins from the current window (callbacks ensure temporal separation)

### Benefits
* Eliminates data leakage: Clear temporal separation between training/validation
* Better generalization: Model validated on unseen time periods
* Standard ML practice: Aligns with GraphCast and GraphDOP

### Trade-offs
Now the validation set requires its own data loading step:
* Original: Validation loaded alongside training (1 data loading step)
* Proposed: Validation requires separate data loading (2 data loading steps per epoch)

### Open Questions
**Validation Frequency**: Currently we run validation every (mini-)epoch. Should we run validation less frequently?
* Consideration: With 3 months of data using 2-week windows, we have ~6 possible training windows. Should we run validation every ~6 epochs (after the model has seen most training data at least once)?
Benefits: Reduced computational overhead, potentially fixes early plotting issues
Notes: This would be about the same as standard monitoring frequency (validation after seeing most training data)

**Comment by mlaurako** [comment]
changes updated in `feature/train-val2`. The `random` callback is tested; however, I have not yet tested or updated the `sequential` callback.

**Comment by mlaurako** [comment]
[rebased] train-val2 performance

100 epochs
runtime: 13h

<img width="3000" height="900" alt="Image" src="https://github.com/user-attachments/assets/3dbb6999-b0af-467a-9c29-0528fbe7512d" />

<img width="3000" height="900" alt="Image" src="https://github.com/user-attachments/assets/9870c299-df5a-4b93-9fc3-f22ad89c17f4" />

<img width="3000" height="900" alt="Image" src="https://github.com/user-attachments/assets/b94a80e7-ccd3-450b-9c79-0d2e9a19fc33" />

<img width="2947" height="772" alt="Image" src="https://github.com/user-attachments/assets/64d9ef70-ae62-46c6-84a6-bbd556fa5c50" />

<img width="2947" height="772" alt="Image" src="https://github.com/user-attachments/assets/cd7f82d5-176b-4923-aa77-5055318bece1" />

<img width="2946" height="772" alt="Image" src="https://github.com/user-attachments/assets/e50e5780-7a82-4ae7-bbca-afde7a94743e" />

<img width="2948" height="772" alt="Image" src="https://github.com/user-attachments/assets/9962c40b-4df0-4c0e-a66c-23875f2cff14" />

<img width="2945" height="771" alt="Image" src="https://github.com/user-attachments/assets/0501ff18-5488-40cc-8d63-38f6df395161" />
